cwms-data-api/src/main/java/cwms/cda/api/BasinController.java

```diff
@@ -144,37 +178,149 @@ public void getOne(@NotNull Context ctx, @NotNull String basinId) {
+                    .withOfficeId(basin.getBasinId().getOfficeId())
+                    .build();
+                basinDao.renameBasin(oldLoc, newLoc);
+                ctx.status(HttpServletResponse.SC_OK).json("Updated Location");
+        }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        responses = {
+            @OpenApiResponse(status = STATUS_204, description = "Basin successfully stored to CWMS."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not implemented")
+        },
+        method = HttpMethod.POST,
+        description = "Creates CWMS Basin",
+        tags = {TAG}
+    )
     @Override
     public void create(@NotNull Context ctx) {
-        ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+        DSLContext dsl = getDslContext(ctx);
+        String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) :
+                Formats.JSONV1;
+        ContentType contentType = Formats.parseHeader(formatHeader, Basin.class);
+        ctx.contentType(contentType.toString());
+
+        cwms.cda.data.dto.basin.Basin basin = Formats.parseContent(contentType, ctx.body(),
+                cwms.cda.data.dto.basin.Basin.class);
+
+        String newBasinId = basin.getBasinId().getName();
+
+        if (contentType.getType().equals(Formats.NAMED_PGJSON)) {
+            ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+        } else {
+            cwms.cda.data.dao.basin.BasinDao basinDao = new cwms.cda.data.dao.basin.BasinDao(dsl);
+            basinDao.storeBasin(basin);
+            ctx.status(HttpServletResponse.SC_CREATED).json(newBasinId + " Created");
+        }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = OFFICE, required = true, description = "Specifies the"
+                    + " owning office of the basin to be renamed."),
+            @OpenApiParam(name = DELETE_MODE, required = true, description = "Specifies the delete method used.")
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                    + "parameters did not find a basin."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not "
+                    + "implemented")
+        },
+        description = "Renames CWMS Basin",
+        tags = {TAG}
+    )
     @Override
-    public void delete(@NotNull Context ctx, @NotNull String s) {
-        ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+    public void delete(@NotNull Context ctx, @NotNull String basinName) {
+        DSLContext dsl = getDslContext(ctx);
+        String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) :
+                Formats.JSONV1;
+        ContentType contentType = Formats.parseHeader(formatHeader, Basin.class);
```

Comment:
I don't think you need contentType or accept header for delete

Reply:
Removed and still functions correctly